libs/sqlite3/Makefile

```diff
@@ -94,12 +94,32 @@ define Package/sqlite3-cli/config
   source "$(SOURCE)/Config-cli.in"
 endef
 
+ifeq ($(CONFIG_SQLITE3_LINENOISE),y)
+LINENOISE_COMMIT:=880b94130ffa5f8236392392b447ff2234b11983
+LINENOISE_FILE:=linenoise-$(LINENOISE_COMMIT).tar.xz
+define Download/linenoise
+  PROTO:=git
+  URL:=https://github.com/antirez/linenoise.git
+  SOURCE_VERSION:=$(LINENOISE_COMMIT)
+  MIRROR_HASH:=8258cc2316899ee6b422d7308496f130e60dbe4a803b26ee12583cd7bf8d6c2f
+  FILE:=$(LINENOISE_FILE)
+  SUBDIR:=linenoise
+endef
```

Comment:
Can we make this dependency a host-only package?

Reply:
https://github.com/openwrt/packages/pull/27493#discussion_r2386278896

Reply:
@GeorgeSapkin , I'm not sure I understand. linenoise is not compiled or installed separately, and it's not a separate package either. It's unpacked into the sqlite's source directory and gets compiled into sqlite3-cli using the target toolchain only.

Reply:
Actually build-only package ;)
see [libtorrent](https://github.com/openwrt/packages/blob/master/libs/libtorrent/Makefile) as example

Reply:
@GeorgeSapkin , @1715173329, please review.
Refactoring linenoise into a dedicated package makes sense. Now other packages that might want to use it can do this without downloading it.
I didn't know about such an option, thank you guys for the suggestion.